samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -75,7 +76,8 @@ static K_SEM_DEFINE(sem_broadcast_code_received, 0U, 1U);
 static K_SEM_DEFINE(sem_pa_request, 0U, 1U);
 static K_SEM_DEFINE(sem_past_request, 0U, 1U);
 static K_SEM_DEFINE(sem_bis_sync_requested, 0U, 1U);
-static K_SEM_DEFINE(sem_bis_synced, 0U, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+static K_SEM_DEFINE(sem_stream_connected, 0U, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
+static K_SEM_DEFINE(sem_stream_started, 0U, CONFIG_BT_BAP_BROADCAST_SNK_STREAM_COUNT);
```

Comment:
Looks to me that `sem_stream_connected` was added as a way to do `k_sem_take` on `sem_stream_started`, but without decreasing the semaphore count. Is that correct?

If so, then it seems like a patch solution, rather than an actual solution.

Reply:
Yes, maybe a patch solution. I will try to make a better solution

Reply:
Yeah, because the current solution may end up with the case where the count of `sem_stream_started` is e.g. 3, but the count of `sem_stream_connected` is 0 (as you still take from `sem_stream_connected`).

Reply:
Did the volatile bool big_synced solution, you suggested